Add version to srf datasets

ekluzek opened the pull request:
Add a version attribute to surface datasets that can be checked for compatibility in ctsm.

Fixes #2723

- [x] Update the comments in the glacier code to remove the bit about it being because surface dataset isn't compatible.

ekluzek committed:
Add global attribute write of dataset compatability as a floating point number to surface datasets

ekluzek committed:
Add a check for compatibility based on the version in the surface dataset

ekluzek commented on src/main/surfrdMod.F90:
```diff
@@ -45,6 +46,70 @@ module surfrdMod
+         if (exists) then
+            version = 5.0_r4
+         else
+            call check_var( ncid, 'PCT_GLC_MEC_GIC', exists)
```
This isn't quite right here.

ekluzek replied:
Corrected

ekluzek commented on tools/mksurfdata_esmf/src/mkfileMod.F90:
```diff
@@ -126,6 +127,7 @@ subroutine mkfile_define_atts(pioid, dynlanduse)
     str = 'Community Land Model: CLM5'
     rcode = pio_put_att (pioid, pio_global, 'Source', trim(str))
     rcode = pio_put_att (pioid, pio_global, 'Version', trim(gitdescribe))
+    rcode = pio_put_att (pioid, pio_global, 'Dataset_Version', data_version)
```
The name should have something with compatibility or the like in the name.

ekluzek replied:
In meeting with @slevis-lmwg he suggests we don't need to add compatibility to the name.

ekluzek commented on tools/mksurfdata_esmf/src/mkfileMod.F90:
```diff
@@ -105,6 +105,7 @@ subroutine mkfile_define_atts(pioid, dynlanduse)
     character(len= 10)   :: time               ! temporary
     character(len=  5)   :: zone               ! temporary
     integer              :: rcode
+    real(r4), parameter  :: data_version = 5.3   !  Compatibility version number of the datasets to create
```
Add comments that the version is only updated when changes to surface datasets make them incompatible with previous versions.

ekluzek commented on src/main/surfrdMod.F90:
```diff
@@ -27,6 +27,7 @@ module surfrdMod
   save
   !
   ! !PUBLIC MEMBER FUNCTIONS:
+  public :: surfrd_check_file       ! Check that this surface dataset is compatible
```
Maybe:

surfrd_compatibility_check
surfrd_compat_check?

ekluzek replied:
latter

ekluzek commented on src/main/surfrdMod.F90:
```diff
@@ -45,6 +46,70 @@ module surfrdMod
+         write(iulog,'(a,f3.1,a,f3.1)') 'Dataset version = ', version, ' Version expected = ', expected_version
+         write(iulog,*) errMsg(sourcefile, __LINE__)
+      end if
+      call endrun(msg="ERROR: Incompatble surface dataset")
```
Fix spelling.

slevis-lmwg reviewed (approved):
@ekluzek and I looked this over, and I'm approving.

ekluzek committed:
Remove error about surface datasets

Remove the error messaging that talks about
old/incompatible surface dataset because we
now are checking for the surface dataset compatability.

ekluzek committed:
Add comments about dataset verison

Add some notes that dataset version is only
updated when the new datasets will NOT be
backwards compatible with the previous
version.

ekluzek committed:
Change name of the subroutine for the surface dataset compatibility check, fix version guessing inside it, fix spelling error, respond to rest of the review conversations
